src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitService.kt

```diff
@@ -144,16 +147,23 @@ class VisitService(
     visitRepository.deleteAllByReferenceIn(visits.map { it.reference }.toList())
   }
 
-  fun cancelVisit(reference: String): VisitDto {
-    log.info("Updating visit for $reference")
+  fun cancelVisit(reference: String, cancelOutcome: OutcomeDto): VisitDto {
+    log.info("Updating visit for $reference with $cancelOutcome")
 
     val visitEntity = visitRepository.findByReference(reference)
     visitEntity ?: throw VisitNotFoundException("Visit reference $reference not found")
 
     visitEntity.visitStatus = VisitStatus.CANCELLED
 
-    // TODO: VB-667 Set Reason [VISIT_OUTCOMES, STATUS_CHANGED_REASON]
+    val visitOutCome = createVisitNote(visitEntity, VISIT_OUTCOMES, cancelOutcome.outcome.name)
```

Comment:
Do we need something like this before this line to prevent 500 exceptions?

    if(visitEntity.visitNotes.any{it.type == VISIT_OUTCOMES})
      throw ValidationException("Visit ${visitEntity.reference} already contains OUTCOME")

Reply:
interesting point! have been wondering about this! what was the constraint,... If anything we should just remove an existing one,... theres only one out come right... it's been canceled!

Reply:
`    val visitOutCome = createVisitNote(visitEntity, VISIT_OUTCOMES, cancelOutcome.outcome.name)
    if (visitEntity.visitNotes.any { it.type == VISIT_OUTCOMES }) {
      log.warn("Removed old outcomes, as there is only one outcome when you cancel a visit!")
      visitEntity.visitNotes.removeAll { it.type == VISIT_OUTCOMES }
    }`

Reply:
Decided to let the normal constraint in the DB layer handle this,.. Otherwise we are adding layers!